src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -1172,9 +1172,7 @@ def attach_managed_data_disk(cmd, resource_group_name, vm_name, disk, new=False,
 
     # pylint: disable=no-member
     if lun is None:
-        luns = ([d.lun for d in vm.storage_profile.data_disks]
-                if vm.storage_profile.data_disks else [])
-        lun = max(luns) + 1 if luns else 0
+        lun = _get_disk_lun(vm.storage_profile.data_disks)
```

Comment:
What if there are no more free luns? 

Should we add some logic to raise a CLIError? Otherwise, the client will get the error "disk lun out of range" instead of something more meaningful like: "no more available disk luns", "No more disks can be attached to this VM" or whatever error message might be more appropriate.

Reply:
We never know how many luns are allowed. For now it is likely 64, but it may be improved to 128 tomorrow. This change is just do the right thing at our end and then leave the rest validation for service. The worst case is service end will reject, but that is fine